packages/std/src/types.rs

```diff
@@ -24,8 +27,13 @@ pub struct TransactionInfo {
     /// using the pair (`env.block.height`, `env.transaction.index`).
     ///
     pub index: u32,
+
+    /// Checksum of the transaction.
+    pub checksum: Checksum,
```

Comment:
The `Checksum` type is made for `hash(wasm_blob)` in CosmWasm. Using it here might be a bit too restrictive. CombetBFT uses 32 bytes sha256 hashes for transactions but this might change and we could see CosmWasm on top of other consensus engines that use different hash sizes.

Because of that I would make this 
```suggestion
    pub hash: Binary,
```

This has three advantages:
- No need to worry about correct string encoding (CometBFT typically uses upper case hex)
- Any hash size works, CosmWasm does not care
- We can use `#[serde(default)]` to set this to empty for pre-3.0 hosts. This is something we already do in `SubMsgResponse`

Reply:
Changed from `Checksum` to `Binary` and added a note about pre-3.0 blockchains in the docstring